### PR TITLE
[ci] move core/serverless window python tests to rayci

### PR DESCRIPTION
**Pull request by can-anyscale**
Move core+serverless window python tests to rayci

Test:
- CI

### CODE REVIEW

File: ci/ray_ci/core.tests.yml [hunk 1]
```diff
@@ -1,3 +1,11 @@
...
   - //python/ray/tests:test_autoscaler
   - windows://:metric_exporter_grpc_test
+  - windows://python/ray/tests:test_actor_retry
+  - windows://python/ray/tests:test_object_spilling
+  - windows://python/ray/tests:test_object_spilling_asan
+  - windows://python/ray/tests:test_object_spilling_debug_mode
+  - windows://python/ray/tests:test_placement_group_3
+  - windows://python/ray/tests:test_reference_counting_2 
+  - windows://python/ray/tests:test_runtime_env_working_dir_3
+  - windows://python/ray/tests:test_task_events_2
```
Comment (aslonnie) [inline ci/ray_ci/core.tests.yml:11]: cc @rickyyx @rkooo567

File: python/ray/tests/BUILD [hunk 2]
```diff
@@ -65,6 +61,18 @@ py_test_module_list(
...
+  files = [
+    "test_gcs_fault_tolerance.py",
+    "test_global_gc.py",
+    "test_job.py",
+    "test_memstat.py",
+  ],
+  size = "medium",
+  tags = ["exclusive", "medium_size_python_tests_a_to_j", "no_windows", "team:core"],
+  deps = ["//:ray_lib", ":conftest"],
+)
```
Comment (aslonnie) [inline python/ray/tests/BUILD:74]: why are these no windows? was this defined in the no windows list in the `ci/ci.sh`?

**Review by aslonnie — APPROVED** [review]

File: python/ray/tests/BUILD (hunk 2 above)
Comment (can-anyscale) [reply python/ray/tests/BUILD:74]: yes, they are from this list https://github.com/ray-project/ray/blob/master/ci/ci.sh#L224

**Review by rickyyx — COMMENTED** [review]
Awesome! 

One question: are we able to run all the new windows tests for like 10 times and see if there's any additional flaky tests show up so we know it's because of the env change?

**Comment by can-anyscale** [comment]
@rickyyx good point, let me maybe run it 5 times on this PR and let postmerge run takes care of the rest; main reason for 5 vs. 10 is because of my own dev velocity and cost (will take 1 hour each to re-run the suite)

**Comment by rickyyx** [comment]
> @rickyyx good point, let me maybe run it 5 times on this PR and let postmerge run takes care of the rest; main reason for 5 vs. 10 is because of my own dev velocity and cost (will take 1 hour each to re-run the suite)

Sure - Could we hack the pipeline yaml a bit so that it automatically runs X times? e.g. random thoughts, like what if we duplicate the steps 10 times with wait in between the steps?

**Comment by can-anyscale** [comment]
@rickyyx good point, let me clone the yml blob

**Comment by can-anyscale** [comment]
@rickyyx all test passed, except for one `windows://python/ray/tests:test_task_metrics` which has 7% of yellowness on go/flaky. Are you ok with giving this the benefit of the doubt to merge this as it is and see how the test behave in postmerge. Thankkks

Screenshot 2023-12-18 at 1 41 13 PM

**Review by rickyyx — APPROVED** [review]
can we run 5 more rounds?

**Comment by can-anyscale** [comment]
let's do it!

**Comment by can-anyscale** [comment]
Everything is passing except for one test (another one) again. Let's merge this and continue to monitor in postmerge. We'll need to continue to adjust.

**Pull request merged**